src/Routing.jl

```diff
@@ -160,29 +162,24 @@ function update_state_from_channel!(
     all_basin_ids = static_env.basin_ids
     attributes_df = static_env.attributes
 
+    start_date = date_window.start_date
+    end_date = date_window.end_date
+    dates = get_all_dates(date_window)
+
     C, D = channel_model.wave_velocity, channel_model.diffusivity
     t_max = channel_model.t_max
 
-
     # Iterate over basins in the given routing level
+    new_state = Dict(eachrow([all_basin_ids zeros(length(all_basin_ids))])) # id -> 0.0  
```

Comment:
Is is straightforward in this PR to update the current state in place? can def be a follow-on PR. but at the end of the day, we set new_state[basin_id] = streamflow, so it seems like this could be state[basin_id] .= streamflow

Reply:
I was using the states to compare to the ref solution on a previous PR. But I agree that we should have a default as overwriting, with an infrequent copy where necessary. I'll look into this.

Reply:
Note created in PR message